### PR TITLE
1098 add ca into macos os store

### PR DESCRIPTION
**Pull request by Thykof**

### CODE REVIEW

File: installer/macos/scripts/postinstall [hunk 1]
```diff
@@ -50,3 +50,5 @@ mkdir -m 777 -p $MASSASTATION_CERT_DIR || fatal "certs directory creation failed
 ping -c 1 -t 1 test.massa >/dev/null 2>&1 || set_local_dns
 
 mv ./uninstall.sh $MASSASTATION_CONFIG_DIR || fatal "uninstall script move failed."
+
+$2/MassaStation.app/Contents/MacOS/massastation --repair || fatal "failed to repair MassaStation"
```
Comment (Thykof) [inline installer/macos/scripts/postinstall]:
```suggestion
LOG_LEVEL=DEBUG $2/MassaStation.app/Contents/MacOS/massastation --repair || fatal "failed to repair MassaStation"
```

File: pkg/certificate/store/store_darwin.go [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+		return nil, fmt.Errorf("failed to unmarshall plistData: %w", err)
+	}
+
+	var plistRoot map[string]interface{}
+
+	_, err = plist.Unmarshal(plistData, &plistRoot)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshall plistData: %w", err)
+	}
+	return plistRoot, nil
```
Comment (Thykof) [inline pkg/certificate/store/store_darwin.go:127]:
```suggestion

	return plistRoot, nil
```

File: pkg/certificate/store/store_darwin.go [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+
+	trustSettings, err := createCertTrustSettings()
+	if err != nil {
+		return fmt.Errorf("failed to create trust settings: %w", err)
+	}
+
+	trustList := plistRoot["trustList"].(map[string]interface{})
+
+	for key := range trustList {
+		logger.Debug(key)
```
Comment (Thykof) [inline pkg/certificate/store/store_darwin.go:168]:
```suggestion
```

**Review by Thykof — COMMENTED** [review]
pkg/certificate/store/store_darwin.go:127:2: return statements should not be cuddled if block has more than two lines (wsl)
	return plistRoot, nil
	^
pkg/certificate/store/store_darwin.go:146:2: return statements should not be cuddled if block has more than two lines (wsl)
	return nil
	^
pkg/certificate/store/store_darwin.go:151:5: type assertion must be checked (forcetypeassert)
	if plistRoot["trustVersion"].(uint64) != 1 {
	   ^
pkg/certificate/store/store_darwin.go:165:2: type assertion must be checked (forcetypeassert)
	trustList := plistRoot["trustList"].(map[string]interface{})
	^
pkg/certificate/store/store_darwin.go:169:3: type assertion must be checked (forcetypeassert)
		entry := trustList[key].(map[string]interface{})
		^
int/api/server.go:144:2: ineffectual assignment to nodeVersion (ineffassign)
	nodeVersion := "unknown"
	^

**Comment by Thykof** [comment]
issue: during the installation, we initiate the repair mode. It writes to the log file, and as a result, it's root that writes the initial log entry. Consequently, when we subsequently launch in normal mode, it becomes impossible to write to the file

File: pkg/certificate/store/store_darwin.go [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+	"bytes"
+	"crypto/x509"
+	"encoding/asn1"
+	"encoding/pem"
+	"fmt"
+	"io/fs"
+	"os"
+	"os/exec"
+
+	"github.com/massalabs/station/pkg/logger"
```
Comment (thomas-senechal) [inline pkg/certificate/store/store_darwin.go:15]:
```suggestion
```

File: cmd/massastation/main.go [hunk 5]
```diff
@@ -63,6 +71,10 @@ func main() {
 		logger.Fatalf("Error with you current system configuration: %s", err.Error())
 	}
 
+	if startFlags.Repair {
+		os.Exit(0)
```
Comment (gregLibert) [inline cmd/massastation/main.go:75]: What about adding a message informing the enduser that  the execution was finished.

File: cmd/massastation/main.go [hunk 6]
```diff
@@ -32,13 +39,14 @@ func ParseFlags() api.StartServerFlags {
...
-	flag.BoolVar(&flags.Version, "version", false, "Print version info and exit")
+	flag.IntVar(&serverFlags.Port, "http-port", httpPort, "HTTP port to listen to")
+	flag.IntVar(&serverFlags.TLSPort, "https-port", httpsPort, "HTTPS port to listen to")
+	flag.BoolVar(&startFlags.Version, "version", false, "Print version info and exit")
+	flag.BoolVar(&startFlags.Repair, "repair", false, "Repair MassaStation")
 
 	flag.Parse()
 
-	return flags
+	return serverFlags, startFlags
```
Comment (gregLibert) [inline cmd/massastation/main.go:49]: startFlag is unclear where serverFlag is. Can you find another terminology?

File: installer/macos/scripts/postinstall [hunk 7]
```diff
@@ -50,3 +50,5 @@ mkdir -m 777 -p $MASSASTATION_CERT_DIR || fatal "certs directory creation failed
 ping -c 1 -t 1 test.massa >/dev/null 2>&1 || set_local_dns
 
 mv ./uninstall.sh $MASSASTATION_CONFIG_DIR || fatal "uninstall script move failed."
+
+LOG_LEVEL=DEBUG $2/MassaStation.app/Contents/MacOS/massastation --repair || fatal "failed to repair MassaStation"
```
Comment (gregLibert) [inline installer/macos/scripts/postinstall:54]: Are you sure debug is needed? Can it disturb the enduser?

File: pkg/certificate/store/store.go [hunk 8]
```diff
@@ -1,5 +1,7 @@
 package store
 
-import "fmt"
+import (
+	"fmt"
+)
```
Comment (gregLibert) [inline pkg/certificate/store/store.go:5]: Why ? is it done by fmt?

File: pkg/certificate/store/store_darwin.go [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+)
+
+var _ runner.Runner = &SecurityRunner{}
+
+const (
+	permissionUrwGrOr = 0o644
+	permissionUrw     = 0o600
+
+	tmpTrustSettingsFile = "trust-settings.plist"
+	tmpTrustedCertFile   = "trusted-cert"
```
Comment (gregLibert) [inline pkg/certificate/store/store_darwin.go:26]: not sure that the temporary notion brings a significant piece of information here.

File: pkg/certificate/store/store_darwin.go [hunk 10]
```diff
@@ -0,0 +1,223 @@
...
+)
+
+// SecurityRunner encapsulates security commands.
+type SecurityRunner struct {
```
Comment (gregLibert) [inline pkg/certificate/store/store_darwin.go:30]: Why security? I think it's a too generic word. It is also a loaded word IMHO.
I suggest to use simply CommandRunner or Runner.

File: pkg/certificate/store/store_darwin.go [hunk 11]
```diff
@@ -0,0 +1,223 @@
...
+		<key>kSecTrustSettingsPolicy</key>
+		<data>
+		KoZIhvdjZAEC
+		</data>
+		<key>kSecTrustSettingsPolicyName</key>
+		<string>basicX509</string>
+		<key>kSecTrustSettingsResult</key>
+		<integer>1</integer>
+	</dict>
+</array>
```
Comment (gregLibert) [inline pkg/certificate/store/store_darwin.go:214]: What about moving all that in trust_darwin.plist file ?

**Review by gregLibert — CHANGES_REQUESTED** [review]

File: installer/macos/scripts/postinstall (hunk 7 above)
Comment (Thykof) [reply installer/macos/scripts/postinstall:54]: The end user in the nominal scenario won't see the logs. I chose to put DEBUG level here to ease debugging when we will potentially ask for the logs to a user.

In the repair process, many logs are in DEBUG level.

File: pkg/certificate/store/store_darwin.go (hunk 10 above)
Comment (Thykof) [reply pkg/certificate/store/store_darwin.go:30]: the name of the binary is `security`. So I named it the same as CertutilRunner.

**Comment by Thykof** [comment]
end 2 end test: work as expected

File: cmd/massastation/main.go [hunk 12]
```diff
@@ -63,6 +62,11 @@ func main() {
 		logger.Fatalf("Error with you current system configuration: %s", err.Error())
 	}
 
+	if stationStartFlags.Repair {
+		logger.Infof("MassaStation has been repaired")
```
Comment (gregLibert) [inline cmd/massastation/main.go:66]:
```suggestion
		logger.Infof("Repair process completed. Please check the logs for any potential errors.")
```

**Review by gregLibert — CHANGES_REQUESTED** [review]

**Review by gregLibert — APPROVED** [review]

**Pull request merged**